### ISSUE TITLE
[RFE] Improve CRD handling of Helm Charts

### ISSUE THREAD

**Issue by mqasimsarfraz** [inspektor-gadget/inspektor-gadget#1814]
As mentioned in the [comment](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/charts/Makefile#L28) helm doesn't handle CRD upgrades well, so we either need to document or improve how helm handle the CRDs. As the [comment](https://github.com/inspektor-gadget/inspektor-gadget/pull/1707#discussion_r1247932775) here suggests we can simply move our CRD to [templates](https://github.com/inspektor-gadget/inspektor-gadget/tree/main/charts/gadget/templates) directory to make this work but it needs to be tested and investigated.

According to https://github.com/helm/helm/issues/8668#issuecomment-683925646:

> If you're chart doesn't install custom resources just the CRDs... you can put the CRDs right into the templates directory.

**Comment by mqasimsarfraz** [comment]
Seems like CRD with helm charts are indeed complicated [[1]](https://github.com/helm/helm/issues/10585#issue-1108447428) [[2]](https://github.com/helm/community/blob/f9e06c16d89ccea1bea77c01a6a96ae3b309f823/architecture/crds.md) but I also think our use-case is simple (w/o CRs, using [ConfigMaps in future](https://github.com/flyth/inspektor-gadget/tree/michael/persistency)) so I will propose to move forward:

- Move the charts to from `crds` to `templates` directory with an option `skipCRDs` (default: false) simplifying upgrades.
- Document the workflow to use `skipCRDs=false` with `kubectl apply ...`, similar to [cert-manager docs.](https://cert-manager.io/docs/installation/helm/) in case user want to be careful with CRs.
- Document the uninstall section with CRs removal to ensure we don't run into finalizers issue. 

@mauriciovasquezbernal what do you think?

**Comment by mauriciovasquezbernal** [comment]
@mqasimsarfraz do you think it's still relevant now that we decided to use configmaps for [headless gadgets](https://www.inspektor-gadget.io/docs/v0.37.0/reference/headless)?

**Comment by mqasimsarfraz** [comment]
Not relevant anymore but we still have handling installing CRDs via helm-charts and kubectl gadget deploy. Should that be removed with buildt-in gadgets?

**Comment by mauriciovasquezbernal** [comment]
Yes, I think all of that will go away with the removal of built-in gadgets.

**Comment by mqasimsarfraz** [comment]
Sounds good, I will close this issue then!
